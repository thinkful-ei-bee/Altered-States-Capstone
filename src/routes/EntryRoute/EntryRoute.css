.entry-charts-entry-container {
  margin-top: 70px;
}

hr {
  width: 90%;
}

.entry-container {
  display: grid;
  grid-template-columns: 10% 40% 40% 10%;
  grid-template-rows: 15px 1fr 40px;
}

.entry-text {
  grid-column: 2/3;
  grid-row: 2/3;
  text-align: left;
  font-size: 1.5em;
  /* margin: auto; */
  margin-top: 25px;
  margin-left: 10%;
  /* margin-right: 5%; */
}

.entry-text-solo {
  grid-column: 2/4;
  grid-row: 2/3;
  text-align: left;
  font-size: 1.5em;
  margin-left: 5%;
  margin-right: 5%;
}

.entry-selfie {
  grid-column: 3/4;
  grid-row: 2/3;
  width: 50%;
  margin: auto;
  margin-top: 25px;
  /* border: black solid 1px; */
  border-radius: 10px;
}

.delete-button {
  grid-column: 3/4;
  grid-row: 3/4;
  background-color: white;
  text-align: right;
}

.fa-trash {
  font-size: 2em;
  opacity: .5;
  margin-right: 10px;
  margin-bottom: 20px;
}

.fa-trash:hover {
  cursor: pointer;
  opacity: 1;
}

@media only screen and (max-width: 760px) {

  .entry-container {
    grid-template-columns: 50% 50%;
  }

  .entry-text {
    font-size: 1em;
    grid-column: 1/2;
  }

  .entry-text-solo {
    font-size: 1em;
    grid-column: 1/3;
  }

  .entry-selfie {
    grid-column: 2/3;
  }

  .delete-button {
    grid-column: 2/3;
    grid-row: 3/4;
  
  }
}



