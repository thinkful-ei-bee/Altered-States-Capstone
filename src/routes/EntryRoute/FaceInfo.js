import React from 'react'
import { Radar, RadarChart, PolarGrid, PolarAngleAxis } from 'recharts'

export default class FaceInfo extends React.Component {
  
  generateData() {
    const { anger, contempt, disgust,
      fear, happiness, neutral, sadness, surprise } = this.props

      const entryEmotions = { anger, contempt, disgust,
      fear, happiness, neutral, sadness, surprise }

      if (!entryEmotions) return

      let data = []

      for (let key in entryEmotions) {
        let amount
        if (entryEmotions[key]) {
          amount = entryEmotions[key] + 10
        }

        else {
          amount = 10
        }
        data.push({
          name: key, amount
        })
      }

    return data
  }
  
  render() {
    const data = this.generateData()
    return (
      <section className='emotion-feedback'>
        <h2 className='chart-title'>Face Analysis</h2>
        <div>
          <RadarChart cx={175} cy={175} outerRadius={50} width={350} height={300} data={data}>
            <PolarGrid />
            <PolarAngleAxis dataKey="name" />
   
            <Radar name="ToneRadar" dataKey="amount" stroke="#8884d8" fill="#8884d8" fillOpacity={0.6}/>
          </RadarChart>
        </div>
      </section>
    )
  }
}