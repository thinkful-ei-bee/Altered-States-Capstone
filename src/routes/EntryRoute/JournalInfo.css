.journal-info .recharts-wrapper {
  margin: auto;
}

.pie-container {
  margin: auto;
}