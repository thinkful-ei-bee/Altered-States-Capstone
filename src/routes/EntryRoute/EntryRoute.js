import React, { Component } from "react";
import "./EntryRoute.css";
import EntryService from "../../services/entry-service";

import EntryCharts from '../../components/EntryCharts/EntryCharts';
import DeleteBox from '../../components/DeleteBox/DeleteBox';


class EntryRoute extends Component {

  constructor(props) {
    super(props)
   
    this.state = {
      id: null,
      user_id: null,
      date_created: null,
      face_url: '',
      happiness: 0,
      face_anger: 0,
      face_contempt: 0,
      face_disgust: 0,
      face_fear: 0,
      face_happiness: 0,
      face_neutral: 0,
      face_sadness: 0,
      face_surprise: 0,
      tone_analytical: 0,
      tone_anger: 0,
      tone_confident: 0,
      tone_fear: 0,
      tone_joy: 0,
      tone_sadness: 0,
      tone_tentative: 0,

      deleting: false
    }
  }

  componentDidMount() {
    const { id } = this.props.match.params

    EntryService.getEntryById(id)
      .then(res => {
        this.setState({...res})
      })
  }

  renderSelfie() {
    return this.state.face_url ? <img className='entry-selfie' src={this.state.face_url} alt='selfie'/> : ''
  }

  handleDelete() {
    this.setState({deleting: true})
  }

  cancelDelete() {
    this.setState({deleting: false})
  }

  async callDelete() {
    if (!this.state.face_url){
      await EntryService.deleteEntry(this.state.id)
    }
    else {
      await EntryService.deleteEntry(this.state.id)
      await EntryService.deleteSelfie(this.state.face_url)
    }
  }

  confirmDelete() {
    this.callDelete()
      .then(() => {
        this.props.history.push('/')
      })
  }



  render() {

    return (
      <div>
        {this.state.deleting && <DeleteBox cancelDelete={this.cancelDelete.bind(this)} confirmDelete={this.confirmDelete.bind(this)}/>}

        <div className='entry-charts-entry-container'>
          <EntryCharts entry={this.state} deleteEntry={this.handleDelete.bind(this)}/>
        </div>

        <hr className='divider' />
        
        <div className='entry-container'>
          {this.renderSelfie()}
          <p className={this.state.face_url ? 'entry-text' : 'entry-text-solo'}>
            {this.state.text}
          </p>
        </div>

      </div>
    );
  }
}

export default EntryRoute;
