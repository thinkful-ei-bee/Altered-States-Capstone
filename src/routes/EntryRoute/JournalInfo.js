import React from 'react'
import {PieChart, Pie, Sector, Cell} from 'recharts'
import './JournalInfo.css'

export default class JournalInfo extends React.Component {
  
  render() {
    const { currentEntry, entryTones } = this.props;

    const COLORS = ['#0088FE', '#00C49F', '#FFBB28', '#FF8042'];
    const RADIAN = Math.PI / 180;

    const renderCustomizedLabel = ({ cx, cy, midAngle, innerRadius, outerRadius, percent, index }) => {
      const radius = innerRadius + (outerRadius - innerRadius) * 0.5;
      const x  = cx + radius * Math.cos(-midAngle * RADIAN);
      const y = cy  + radius * Math.sin(-midAngle * RADIAN);
    
      return (
        <text x={x} y={y} fill="white" textAnchor={x > cx ? 'start' : 'end'} 	dominantBaseline="central">
          {`${(percent * 100).toFixed(0)}%`}
        </text>
      );
    };

    const data = [
      {name: 'Group A', value: 400}, 
      {name: 'Group B', value: 300},
      {name: 'Group C', value: 300}, 
      {name: 'Group D', value: 200}
    ];
    return (
      <section className='journal-feedback'>
        <p>{currentEntry}</p>
        <div className='pie-container'>
            <PieChart width={205} height={205} onMouseEnter={this.onPieEnter}>
                <Pie
                data={data} 
                cx={100} 
                cy={100} 
                labelLine={false}
                label={renderCustomizedLabel}
                outerRadius={80} 
                fill="#8884d8">
                    {data.map((entry, index) => <Cell fill={COLORS[index % COLORS.length]}/>)}
                </Pie>
            </PieChart>
        </div>
      </section>


    )
  }
}