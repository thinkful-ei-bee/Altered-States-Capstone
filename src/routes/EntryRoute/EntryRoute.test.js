import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom'
import EntryRoute from './EntryRoute';
import toJson from 'enzyme-to-json'
import { shallow, mount } from 'enzyme'
import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({ adapter: new Adapter() })

describe('DashboardRoute', () => {

  it('renders the complete page', () => {
    const wrapper = shallow(<EntryRoute />);
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})