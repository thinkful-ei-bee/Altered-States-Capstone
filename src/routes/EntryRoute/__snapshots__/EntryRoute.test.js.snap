// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardRoute renders the complete page 1`] = `
<div>
  <div
    className="entry-charts-entry-container"
  >
    <EntryCharts
      entry={
        Object {
          "date_created": null,
          "deleting": false,
          "face_anger": 0,
          "face_contempt": 0,
          "face_disgust": 0,
          "face_fear": 0,
          "face_happiness": 0,
          "face_neutral": 0,
          "face_sadness": 0,
          "face_surprise": 0,
          "face_url": "",
          "happiness": 0,
          "id": null,
          "tone_analytical": 0,
          "tone_anger": 0,
          "tone_confident": 0,
          "tone_fear": 0,
          "tone_joy": 0,
          "tone_sadness": 0,
          "tone_tentative": 0,
          "user_id": null,
          "zoom": false,
        }
      }
      selfie={false}
    />
  </div>
  <hr
    className="divider"
  />
  <div
    className="entry-container"
  >
    <div
      className="text-selfie-div"
    >
      <p
        className="entry-text"
      />
    </div>
    <div
      className="entry-buttons"
    >
      <HashLink
        className="back-button"
        to="/#entry-null"
      >
        <i
          className="fa fa-arrow-left"
        />
      </HashLink>
      <i
        className="fa fa-trash"
        onClick={[Function]}
      />
    </div>
  </div>
</div>
`;
